ui/src/containers/AlertProvider.js

```diff
@@ -0,0 +1,32 @@
+import React, { createContext, useContext } from 'react';
+import { useQuery } from 'react-query';
+import { getAlerts } from '../services/alertmanager/api';
+import { filterAlerts } from '../services/alertUtils';
+
+const AlertContext = createContext(null);
+
+export function useAlerts(filters: FilterLabels) {
+  const query = useContext(AlertContext);
+  if (query.status === 'success') {
+    return {
+      ...query,
+      data: filterAlerts(query.data, filters),
+    };
+  }
+  return query;
+}
+
+const AlertProvider = ({ children }) => {
+  const query = useQuery('activeAlerts', () =>
+    getAlerts().then((data) => {
+      return data;
+    }),
+  );
```

Comment:
It should keep the data updated by refetching the alerts frequently (https://react-query.tanstack.com/examples/auto-refetching).

```suggestion
  const query = useQuery('activeAlerts', () => getAlerts(), {
      // Refetch the alerts every 10 second
      refetchInterval: 10000,
    });
```